convert-hf-to-gguf.py

```diff
@@ -211,6 +213,59 @@ def from_model_architecture(model_architecture):
             return MiniCPMModel
         if model_architecture == "BertModel":
             return BertModel
+
+    @staticmethod
+    def from_model_name(model_name: str):
```

Comment:
Was this ever used? And why this function duplicated below?

Reply:
> Was this ever used? And why this function duplicated below?

I'm not sure. Looks like a convenient function for mapping if someone needs it in future convert-hf-to-gguf work.  (The duplication is likely my fault, of course).
Should I get rid of it, comment it out?  (Remember, I'm just merging their contributed deepcode/hf tokenizer code over mostly blindly -- although it does work.  Resolves that out-of-range error too.) @ggerganov.

Let me know -- I'll also have to rebase and resubmit.

Reply:
I just checked - it looks like ggerganov accidentally dropped this in [`d24da31` (#4070)](https://github.com/ggerganov/llama.cpp/pull/4070/commits/d24da31d2f37e2417f038fd7b46b1d8e5ad684b4). It's apparently used for forcing the model used via the command line? This will really be out-of-place after #5825, it should probably just be removed unless @DOGEwbx ever comes back to explain what it's for.

Reply:
I see now - DeepseekCoderModel and DeepseekLLMModel can't be disambiguated from the model architecture alone. This should be changed so that they use a single class that derives tokenizer_model from either the model's config, or the command-line arguments if it really is a user choice.